### DISCUSSION TITLE
What is the reason for Environment::add_template_owned() being fallible?

### DISCUSSION THREAD

**Discussion by jaskij** [mitsuhiko/minijinja#634] [Q&A] [answered]
For unrelated reasons, I'm refactoring a piece of code using minijinja, and what caught my attention is that `Environment::add_template_owned()` is fallible. What is the reason for this? Is it just that the key could be repeated?

I'm asking because it *looks* like something that should be an invariant in my use case (an Environment with exactly one template, inserted during struct construction), and as such it should be handle with `expect()` rather than a fallible function. But without documentation on this, I can not proceed.

**Comment by mitsuhiko** [comment, marked as answer]
I improved the documentation.  The method will fail if there is a syntax error in the template.

https://github.com/mitsuhiko/minijinja/commit/1ce7991d5a1cd1b3865d41673d1de04339c9a48f

**Reply by jaskij** [reply]
Thank you very much.
